src/v0/destinations/delighted/util.js

```diff
@@ -50,53 +74,47 @@ const userValidity = async (channel, Config, userId) => {
   }
 
   const basicAuth = Buffer.from(Config.apiKey).toString('base64');
-  let response;
-  try {
-    response = await myAxios.get(
-      `${ENDPOINT}`,
-      {
-        headers: {
-          Authorization: `Basic ${basicAuth}`,
-          'Content-Type': JSON_MIME_TYPE,
-        },
-        params: paramsdata,
+  const { processedResponse } = await handleHttpRequest(
+    'get',
+    `${ENDPOINT}`,
+    {
+      headers: {
+        Authorization: `Basic ${basicAuth}`,
+        'Content-Type': JSON_MIME_TYPE,
       },
+      params: paramsdata,
+    },
+    {
+      destType: 'delighted',
+      feature: 'transformation',
+      requestMethod: 'GET',
+      endpointPath: '/people.json',
+      module: 'router',
+    },
+  );
+
+  if (!isHttpStatusSuccess(processedResponse.status)) {
+    const errStatus = getErrorStatus(processedResponse.status);
+    throw new NetworkError(
+      `Error occurred while checking user : ${JSON.stringify(processedResponse.response)}`,
+      errStatus,
       {
-        destType: 'delighted',
-        feature: 'transformation',
-        requestMethod: 'GET',
-        endpointPath: '/people.json',
-        module: 'router',
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errStatus),
       },
+      processedResponse,
     );
-    if (response && response.data && response.status === 200 && Array.isArray(response.data)) {
-      return response.data.length > 0;
-    }
-    throw new NetworkInstrumentationError('Invalid response');
-  } catch (error) {
-    let errMsg = '';
-    let errStatus = 400;
-    if (error.response && error.response.data) {
-      errMsg = JSON.stringify(error.response.data);
-      switch (error.response.status) {
-        case 422:
-        case 401:
-        case 406:
-        case 403:
-          errStatus = 400;
-          break;
-        case 500:
-        case 503:
-          errStatus = 500;
-          break;
-        default:
-          errStatus = 400;
-      }
-    }
-    throw new NetworkError(`Error occurred while checking user : ${errMsg}`, errStatus, {
-      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errStatus),
-    });
   }
+
+  if (
+    processedResponse &&
+    processedResponse.response &&
+    processedResponse.status === 200 &&
+    Array.isArray(processedResponse.response)
+  ) {
+    return processedResponse.response.length > 0;
+  }
+
+  throw new NetworkInstrumentationError('Invalid response');
```

Comment:
Should we throw instrumentation error here?

Reply:
Can you check my changes once and confirm if they are looking good ?